src/main/java/org/jenkinsci/plugins/workflow/actions/ErrorAction.java

```diff
@@ -140,7 +151,7 @@ public String getUrlName() {
      */
     public static @CheckForNull FlowNode findOrigin(@NonNull Throwable error, @NonNull FlowExecution execution) {
         FlowNode candidate = null;
-        for (FlowNode n : new ForkScanner().allNodes(execution)) {
+        for (FlowNode n : new DepthFirstScanner().allNodes(execution)) {
```

Comment:
We should probably revert this change, I forgot that the current candidate replacement is currently relying on `ForkScanner` to guarantee that we always visit children before their parents, and `DepthFirstScanner` makes no such guarantee. Whether this would be a real problem I do not know (would have to involve parallel branches all throwing the same error, e.g. `WorkflowRun.doTerm`), but seems preferable to avoid changing it here seeing there are no relevant tests.

Reply:
Right, already done.